pytroll_collectors/tests/test_segments.py

```diff
@@ -607,14 +621,19 @@ def test_publish_service_name(self):
         self.assertEqual(publish_service_name, "segment_gatherer_iodc_msg")
 
     @patch("pytroll_collectors.segments.logging")
-    @patch("pytroll_collectors.segments.glob")
-    def test_check_and_add_existing_files(self, glob, logging):
+    @patch("fsspec.filesystem")
+    def test_check_and_add_existing_files(self, filesystem, logging):
         """Test that existing matching files are added to the slot."""
         existing_files = [
             "/home/lahtinep/data/satellite/geo/msg/H-000-MSG3__-MSG3________-VIS006___-000007___-201611281100-__",
             "/home/lahtinep/data/satellite/geo/msg/H-000-MSG3__-MSG3________-VIS006___-000008___-201611281100-__",
             # A file that should not match
             "/home/lahtinep/data/satellite/geo/msg/H-000-MSG4__-MSG4________-VIS006___-000008___-201611281100-__"]
+        glob = MagicMock()
+        glob.return_value = existing_files
+        fs_ = MagicMock()
+        fs_.glob = glob
+        filesystem.return_value = fs_
         glob.glob.return_value = existing_files
```

Comment:
I'm not sure I understand why both `glob.return_value` and `glob.glob.return_value` need to be set to the same thing.

Also this test is pretty long, could it be split for clarity?

Reply:
The `glob.glob` one is just a mistake, I tend to do `import glob` and `glob.glob(pattern)` so I've thought I'm using `glob.glob` here even if I'm not. Removing that. and having a look at restructuring the test. It should at least be 3 separate tests.

Reply:
Restructured in b0b769843210f35bc22d952d4fbf9d15de99d2bb